src/NuGet.Core/NuGet.Commands/ClientCertificatesCommand/ListClientCertRunner.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using NuGet.Common;
+using NuGet.Configuration;
+
+namespace NuGet.Commands
+{
+    public class ListClientCertRunner
+    {
+        public static void Run(ListClientCertArgs args, Func<ILogger> getLogger)
+        {
+            args.Validate();
+            var settings = RunnerHelper.GetSettings(args.Configfile);
+            var clientCertificateProvider = new ClientCertificateProvider(settings);
+
+            var items = clientCertificateProvider.GetClientCertificates();
+
+            if (!items.Any())
+            {
+                getLogger().LogInformation(Strings.NoClientCertificates);
+                return;
+            }
+
+            var clientCertificatesLogs = new List<LogMessage>();
+
+            getLogger().LogInformation(Strings.RegsiteredClientCertificates);
+            getLogger().LogInformation(Environment.NewLine);
+
+            for (var i = 0; i < items.Count; i++)
+            {
+                var item = items[i];
+
+                var builder = new StringBuilder();
+
+                var index = $" {i + 1,-6}.";
+                const string defaultIndentation = "      ";
+
+                builder.Append(index);
+                builder.AppendFormat(CultureInfo.CurrentCulture, Strings.ClientCertificatesLogTitle, item.PackageSource, item.ElementName);
+                builder.AppendLine();
+
+                if (item is FileClientCertItem fileCertItem)
+                {
+                    builder.AppendLine($"{defaultIndentation}{string.Format(CultureInfo.CurrentCulture, Strings.ClientCertificatesFileCertFilePath, fileCertItem.FilePath)}");
+                    if (string.IsNullOrEmpty(fileCertItem.Password))
+                    {
+                        builder.AppendLine($"{defaultIndentation}{string.Format(CultureInfo.CurrentCulture, Strings.ClientCertificatesFileCertNoPassword)}");
+                    }
+                    else
+                    {
+                        builder.AppendLine($"{defaultIndentation}{string.Format(CultureInfo.CurrentCulture, Strings.ClientCertificatesFileCertWithPassword)}");
+                    }
+                }
+
+                if (item is StoreClientCertItem storeCertItem)
+                {
+                    builder.AppendLine($"{defaultIndentation}{string.Format(CultureInfo.CurrentCulture, Strings.ClientCertificatesStoreCertStoreLocation, StoreClientCertItem.GetString(storeCertItem.StoreLocation))}");
```

Comment:
Avoid concatenating resource strings. Instead, add a placeholder indentation in the string resource.

```suggestion
                    builder.AppendLine(string.Format(CultureInfo.CurrentCulture, Strings.ClientCertificatesStoreCertStoreLocation, defaultIndentation, StoreClientCertItem.GetString(storeCertItem.StoreLocation)));
```

Reply:
Moved indentations to resource placeholder

Reply:
done